pxr/imaging/plugin/hdEmbree/renderer.cpp

```diff
@@ -58,11 +69,118 @@ class _ScopedThreadScheduler {
 
 #endif  // TBB_INTERFACE_VERSION_MAJOR < 12
 
-
 namespace {
 
 PXR_NAMESPACE_USING_DIRECTIVE
 
+// -------------------------------------------------------------------------
```

Comment:
I feel like it might be a kindness to future us if I pull all of the math and sampling code into a separate file?  lightSamplingUtils.h?  Any objections?